.github/workflows/build.yml

```diff
@@ -52,6 +55,58 @@ jobs:
         with:
           name: ${{ matrix.nexus_classifier }}-rpm
           path: RPMS/${{ matrix.target_arch }}/*.rpm
+
+  Test:
+    name: Test on ${{ matrix.base_image }} for ${{ matrix.target_arch }}
+    runs-on: ubuntu-latest
+    needs: [configure,Build]
+    strategy:
+      matrix: ${{ fromJson(needs.configure.outputs.matrix_test) }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set build variables
+        id: variables
+        run: |
+          echo "imagemagick_version=$(cat ./imagemagick-version)" >> $GITHUB_OUTPUT
+          echo "imagemagick_release=$(cat ./release-version)" >> $GITHUB_OUTPUT
+
+      - name: Download RPMs
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ matrix.nexus_classifier }}-rpm
+
+      - name: List RPMs
+        run: ls -R *.rpm
+
+      - name: Create a container
+        run: |
+          docker run --name testenv -d -it ${{ matrix.base_image }}
+
+      - name: Copy RPMs
+        run: |
+          docker cp ImageMagick-libs-${{ steps.variables.outputs.imagemagick_version }}.${{ matrix.target_arch }}.rpm testenv:/
+          docker cp ImageMagick-${{ steps.variables.outputs.imagemagick_version }}.${{ matrix.target_arch }}.rpm testenv:/
+
+      - name: Install RPMs
+        run: |
+          docker exec -t testenv yum install -y epel-release
+          docker exec -t testenv yum install -y ImageMagick-libs-${{ steps.variables.outputs.imagemagick_version }}.${{ matrix.target_arch }}.rpm
+          docker exec -t testenv yum install -y ImageMagick-${{ steps.variables.outputs.imagemagick_version }}.${{ matrix.target_arch }}.rpm
+
+      - name: Test RPMs output
+        run: |
+          echo "ldd output of convert"
+          docker exec -t testenv ldd /usr/bin/convert
+          echo "Testing convert command"
+          docker exec -t testenv convert -version
+          echo "Creating test image file"
+          convert -size 32x32 xc:transparent test.png
+          echo "Converting png to jpg"
+          convert test.png test1.jpg
+
       - name: Deploy to nexus
```

Comment:
at this point I prefer having a dedicated deploy job - feels a bit cheating that we are doing deploy at the end of a job which is called 'test'